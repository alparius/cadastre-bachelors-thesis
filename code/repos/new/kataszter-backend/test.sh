#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
