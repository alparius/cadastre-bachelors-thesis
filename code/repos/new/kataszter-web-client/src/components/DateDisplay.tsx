import React from "react";
import { Popup } from "semantic-ui-react";

import IDate from "../data/Date";
import dateIntoString from "../util/dateConverts/dateIntoString";
import ItalicFM from "../util/styledMessages/ItalicFM";

interface IDateDisplayProps {
    date: IDate;
}

const DateDisplay: React.FC<IDateDisplayProps> = (props: IDateDisplayProps) => {
    const { date } = props;

    const dateText = dateIntoString(date);

    if (dateText) {
        if (date.comment) {
            return <Popup content={date.comment} key={date.comment} trigger={<span>{dateText}</span>} />;
        } else {
            return <>{dateText}</>;
        }
    } else {
        if (date.comment) {
            return <i style={{ color: "gray" }}>{date.comment}</i>;
        } else {
            return <ItalicFM id="general.noData" />;
        }
    }
};

export default DateDisplay;
