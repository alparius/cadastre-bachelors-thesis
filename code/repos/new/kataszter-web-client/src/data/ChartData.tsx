interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
