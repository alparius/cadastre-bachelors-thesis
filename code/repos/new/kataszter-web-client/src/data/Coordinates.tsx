interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
