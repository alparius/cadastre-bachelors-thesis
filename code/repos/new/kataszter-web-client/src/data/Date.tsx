interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
