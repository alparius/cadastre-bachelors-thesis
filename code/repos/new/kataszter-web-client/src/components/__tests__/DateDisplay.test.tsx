/*===================
  DateDisplay.unit.test.tsx
 ===================*/
import React from "react";

import { NewTestDate } from "../../util/test/data/test-data";
import { createComponentWithIntl } from "../../util/test/test-wrapper";
import DateDisplay from "../DateDisplay";

describe("DateDisplay element", () => {
    it("should render a popup with a span, full date given", () => {
        const container = createComponentWithIntl(<DateDisplay date={NewTestDate} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render just a span, no comment given", () => {
        const testDate = NewTestDate;
        testDate.comment = "";
        const container = createComponentWithIntl(<DateDisplay date={testDate} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render just a span with a comment, no date given", () => {
        const testDate = NewTestDate;
        testDate.year = 0;
        testDate.comment = "test";
        const container = createComponentWithIntl(<DateDisplay date={testDate} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a noData label, nothing given", () => {
        const testDate = NewTestDate;
        testDate.year = 0;
        testDate.comment = "";
        const container = createComponentWithIntl(<DateDisplay date={testDate} />).toJSON();
        expect(container).toMatchSnapshot();
    });
});
