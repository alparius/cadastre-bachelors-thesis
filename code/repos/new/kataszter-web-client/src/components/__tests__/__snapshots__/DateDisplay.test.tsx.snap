// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateDisplay element should render a noData label, nothing given 1`] = `
<i>
  No data
</i>
`;

exports[`DateDisplay element should render a popup with a span, full date given 1`] = `
<span
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  2000.06.06
</span>
`;

exports[`DateDisplay element should render just a span with a comment, no date given 1`] = `
<i
  style={
    Object {
      "color": "gray",
    }
  }
>
  test
</i>
`;

exports[`DateDisplay element should render just a span, no comment given 1`] = `"2000.06.06"`;
